components/kyma-environment-broker/internal/storage/driver/postsql/instance.go

```diff
@@ -18,30 +21,50 @@ func NewInstance(sess dbsession.Factory) *Instance {
 
 func (s *Instance) GetByID(instanceID string) (*internal.Instance, error) {
 	sess := s.NewReadSession()
-	instance, err := sess.GetInstanceByID(instanceID)
-	if err != nil {
-		return nil, errors.Wrapf(err, "while getting instance by ID %s", instanceID)
-	}
-
-	return &instance, nil
+	instance := &internal.Instance{}
+	err := wait.Poll(defaultRetryInterval, defaultRetryTimeout, func() (bool, error) {
```

Comment:
tbh I'm not sure if this retry logic should be placed here. IMO it should be in upper layer  or just some wrapper around that storage which will be responsible for retries and with such approach u will also implement that only once instead of copying that retrie functionality for each function. But let's agree that for now it's good enough, and we will change it in the future 

same thing for retries errors: In current implementation we do not retry on "NotFound" error but in general we are eventually consistent system, so doing that on not found error should be also ok.

Reply:
good suggestion, I will add a todo comment